conn.go

```diff
@@ -263,15 +263,19 @@ func (s *Swarm) setupConn(netConn tpt.Conn, isServer bool) (*Conn, error) {
 	defer s.connLock.Unlock()
 
 	// check for it again as it may have been added already. (TOCTTOU)
-	for c := range s.conns {
-		if c.netConn == netConn {
-			return c, nil
-		}
+	if c, ok := s.connByNet[netConn]; ok {
+		s.connLock.RUnlock()
```

Comment:
This should be `Unlock` not `RUnlock`.

Reply:
Yeah, i'd like to see a test that will fail with this bug.

Reply:
(actually, it shouldn't be there at all; we defer the unlock (copy-paste error...)).